app/src/main/java/org/thunderdog/challegram/component/user/UserView.java

```diff
@@ -197,6 +200,18 @@ public void updateName () {
       name = null;
     }
     float availWidth = getMeasuredWidth() - textLeftMargin - offsetLeft - paddingRight - (unregisteredContact != null ? Screen.dp(32f) : 0);
+    if (user != null) {
+      TdApi.Chat rawChat = tdlib.chat(user.getChatId());
```

Comment:
This method is too heavy for the `onDraw` method. The state of `view` should be remembered at the moment `user` value is set.